In [1]:
import pandas as pd

# Read in the data
data = pd.read_csv('beans.csv')

In [2]:
# Drop the 'Genotype' column
data = data.drop(['Class_cat'], axis=1)

In [3]:
from sklearn.model_selection import train_test_split

# Separate the target variable from the features
X = data.drop('Class', axis=1)
y = data['Class']

# Split the data into a train set and a test set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [4]:
from sklearn.ensemble import RandomForestClassifier

# Create a Random Forest Classifier object
rfc = RandomForestClassifier(random_state=42)

# Fit the model on the training data
rfc.fit(X_train, y_train)


RandomForestClassifier(random_state=42)

In [5]:
from sklearn.metrics import accuracy_score

# Make predictions on the training and test data
train_preds = rfc.predict(X_train)
test_preds = rfc.predict(X_test)

# Calculate the accuracy score on the training and test data
train_acc = accuracy_score(y_train, train_preds)
test_acc = accuracy_score(y_test, test_preds)

print('Training accuracy:', train_acc)
print('Test accuracy:', test_acc)


Training accuracy: 1.0
Test accuracy: 0.9261843554902681


# Conclusion and recommendation
The model achieved a high accuracy score of 1.0 on the training data and a score of 0.93 on the test data. This indicates that the model has learned the patterns in the data very well and is able to generalize to new data.

However, we should be cautious in interpreting these results as the dataset is relatively small and may not be representative of all bean varieties. Additionally, we have only used one machine learning algorithm and have not performed any hyperparameter tuning or model selection.

Therefore, it is recommended to further explore the dataset, try different machine learning algorithms, and perform hyperparameter tuning to optimize the model's performance.